app/services/ai_backend/open_ai.rb

```diff
@@ -54,6 +55,14 @@ def stream_handler(&chunk_received_handler)
       content_chunk = intermediate_response.dig("choices", 0, "delta", "content")
       tool_calls_chunk = intermediate_response.dig("choices", 0, "delta", "tool_calls")
 
+      if (input_tokens = intermediate_response.dig("usage", "prompt_tokens"))
+        @message.input_token_count += input_tokens
+      end
+
+      if (output_tokens = intermediate_response.dig("usage", "completion_tokens"))
+        @message.output_token_count += output_tokens
+      end
```

Comment:
This may not be worth simplifying further, but since the two of these are closely related we could conceptually make it a single thought. You decide if this is clearer or not:

```suggestion
      if (input_tokens, output_tokens = intermediate_response["usage"]&.values_at("prompt_tokens", "completion_tokens"))
        @message.input_token_count += input_tokens.to_i
        @message.output_token_count += output_tokens.to_i
      end
```